spec/requests/api/snapshots_spec.rb

```diff
@@ -0,0 +1,184 @@
+      create_time = Time.zone.parse("2017-01-11T00:00:00Z")
+      snapshot = FactoryGirl.create(:snapshot, :vm_or_template => vm, :create_time => create_time)
+
+      run_get("#{vms_url(vm.id)}/snapshots/#{snapshot.id}")
+
+      expected = {
+        "create_time"       => create_time.iso8601,
+        "href"              => a_string_matching("#{vms_url(vm.id)}/snapshots/#{snapshot.id}"),
+        "id"                => snapshot.id,
+        "vm_or_template_id" => vm.id
+      }
+      expect(response.parsed_body).to include(expected)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "will not show a snapshot unless authorized" do
+      api_basic_authorize
+      vm = FactoryGirl.create(:vm_vmware)
+      snapshot = FactoryGirl.create(:snapshot, :vm_or_template => vm)
+
+      run_get("#{vms_url(vm.id)}/snapshots/#{snapshot.id}")
+
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    describe "POST /api/vms/:c_id/snapshots" do
+      it "can queue the creation of a snapshot" do
+        api_basic_authorize(subcollection_action_identifier(:vms, :snapshots, :create))
+        ems = FactoryGirl.create(:ext_management_system)
+        host = FactoryGirl.create(:host, :ext_management_system => ems)
+        vm = FactoryGirl.create(:vm_vmware, :name => "Alice's VM", :host => host, :ext_management_system => ems)
+
+        run_post("#{vms_url(vm.id)}/snapshots", :name => "Alice's snapsnot")
+
+        expected = {
+          "results" => [
+            a_hash_including(
+              "success"   => true,
+              "message"   => "Creating snapshot for ManageIQ::Providers::Vmware::InfraManager::Vm id: #{vm.id} name: Alice's VM",
+              "task_id"   => anything,
+              "task_href" => a_string_matching(tasks_url)
+            )
+          ]
+        }
+        expect(response.parsed_body).to include(expected)
+        expect(response).to have_http_status(:ok)
+      end
+
+      it "renders a failed action response if snapshotting is not supported" do
+        api_basic_authorize(subcollection_action_identifier(:vms, :snapshots, :create))
+        vm = FactoryGirl.create(:vm_vmware)
+
+        run_post("#{vms_url(vm.id)}/snapshots", :name => "Alice's snapsnot")
+
+        expected = {
+          "results" => [
+            a_hash_including(
+              "success" => false,
+              "message" => "The VM is not connected to a Host"
+            )
+          ]
+        }
+        expect(response.parsed_body).to include(expected)
+        expect(response).to have_http_status(:ok)
+      end
+
+      it "will not create a snapshot unless authorized" do
+        api_basic_authorize
+        vm = FactoryGirl.create(:vm_vmware)
+
+        run_post("#{vms_url(vm.id)}/snapshots", :description => "Alice's snapshot")
+
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+
+    describe "POST /api/vms/:c_id/snapshots?action=delete" do
```

Comment:
Maybe update to "POST /api/vms/:c_id/snapshots   with delete action" just to clarify that this is not a URL signature  (i.e. supporting ?action=delete parameter on POSTs).